cypress/integration/dashboard/dashboard_spec.js

```diff
@@ -142,4 +178,66 @@ describe('Dashboard', () => {
       cy.get('@textboxEl').should('contain', newContent);
     });
   });
+
+  describe('Widget', () => {
+    before(function () {
+      cy.login();
+      createNewDashboardByAPI('Foo Bar')
+        .then(slug => `/dashboard/${slug}`)
+        .as('DashboardUrl');
+    });
+
+    beforeEach(function () {
+      cy.visit(this.DashboardUrl);
+    });
+
+    it('adds widget', () => {
+      addWidget().then(({ id }) => {
+        cy.getByTestId(`WidgetId${id}`).should('exist');
+      });
+    });
+
+    it('removes widget', () => {
+      addWidget().then(({ id }) => {
+        cy.getByTestId(`WidgetId${id}`)
+          .within(() => {
+            cy.get('.widget-menu-remove').click();
+          })
+          .should('not.exist');
+      });
+    });
+
+    describe('Auto height for table visualization', () => {
+      it('has correct visualization and position config', () => {
+        addWidget().then(({ options, visualization }) => {
+          expect(visualization.type).to.eq('TABLE');
+          expect(options.position.autoHeight).to.be.true;
+        });
+      });
```

Comment:
This is unnecessary, right? 🤔

Reply:
I think the cost for it now isn't worth it (currently 2 sec). I wonder how long it would be if you created the widget by API.

Reply:
The problem with creating an API call for Widget/Textbox is that it requires some dynamically generated position data to be sent with it:
```js
data = {
  ...
  options: {
    ...
    position: {
      autoHeight: true,
      col: 3,
      maxSizeX: 6,
      maxSizeY: 1000,
      minSizeX: 1,
      minSizeY: 1,
      row: 0,
      sizeX: 3,
      sizeY: 3,
    },
  },
}
```

that I'd rather not have to mimic. We could get away with it if the dashboard is a clean slate, though that has a cost of its own.

Reply:
@ranbena (cc @gabrieldutra) Actually not all that fields are required (some of them are saved accidentally, one day I'm going to fix that):
```javascript
position: {
  autoHeight: true,
  col: 3,
  row: 0,
  sizeX: 3,
  sizeY: 3,
},
```
All that values could be set manually, just check constraints for selected widget type (e.g. not all widgets can have `autoHeight=true`, some have min width/height; position can be arbitrary at all, just keep in mind that widgets "flow" to top if there is free space)

Reply:
Ok I'll go for it.

Reply:
Implemented it. Hope the diff isn't too heavy to review..
Here's my summary:
**Pros**
* -10 seconds run time

**Cons** (imho):
* Since the ui doesn't update after the "add widget" api request was executed , must reload the page for the widget to appear. Had to move the `cy.visit(this.dashboardUrl)` from `beforeEach` to inside the test.
* For the same reason, `addTextboxByAPI` and `addWidgetByAPI` can't return an element reference to be aliased, since the element doesn't yet exist in the dom. I tried to work around that limitation but eventually fell back to returning `elTestId` which will then require the test to `cy.getByTestId(elTestId)`. 
* Currently, settling on hardcoded "first widget" positioning config `{ col: 0, row: 0, sizeX: 3, sizeY: 3 }` but if future test requires multiple widgets, must do calculation (or opt for widget add from ui).

Overall I think this added more "magic" to the tests which I favor for production code and less for testing code as readability and scalability are more important than line count and abstraction. 

Lmk what you think @gabrieldutra and @levko and if you can improve the code.